checks/sast.go

```diff
@@ -205,3 +222,106 @@ func codeQLInCheckDefinitions(c *checker.CheckRequest) (int, error) {
 	})
 	return checker.MinResultScore, nil
 }
+
+type sonarConfig struct {
+	url  string
+	file checker.File
+}
+
+// nolint
+func sonarEnabled(c *checker.CheckRequest) (int, error) {
+	var config []sonarConfig
+	err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
+		Pattern:       "*",
+		CaseSensitive: false,
+	}, validateSonarConfig, &config)
+	if err != nil {
+		return checker.InconclusiveResultScore, err
+	}
+	for _, result := range config {
+		c.Dlogger.Info(&checker.LogMessage{
+			Path:      result.file.Path,
+			Type:      result.file.Type,
+			Offset:    result.file.Offset,
+			EndOffset: result.file.EndOffset,
+			Text:      "Sonar configuration detected",
+		})
+	}
+
+	if len(config) > 0 {
+		return checker.MaxResultScore, nil
+	}
+
+	return checker.MinResultScore, nil
+}
+
+// Check file content.
+var validateSonarConfig fileparser.DoWhileTrueOnFileContent = func(pathfn string,
+	content []byte,
+	args ...interface{},
+) (bool, error) {
+	if path.Base(pathfn) != "pom.xml" {
+		return true, nil
+	}
+
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"validateSonarConfig requires exactly 1 arguments: %w", errInvalid)
+	}
+
+	// Verify the type of the data.
+	pdata, ok := args[0].(*[]sonarConfig)
+	if !ok {
+		return false, fmt.Errorf(
+			"validateSonarConfig expects arg[0] of type *[]sonarConfig]: %w", errInvalid)
+	}
+
+	regex := regexp.MustCompile(`<sonar\.host\.url>\s*(https:\/\/.*)\s*<\/sonar\.host\.url>`)
+	match := regex.FindSubmatch(content)
+
+	if len(match) < 2 {
+		return true, nil
+	}
+
+	offset, err := findLine(content, []byte("<sonar.host.url>"))
+	if err != nil {
+		return false, err
+	}
+
+	endOffset, err := findLine(content, []byte("</sonar.host.url>"))
+	if err != nil {
+		return false, err
+	}
+
+	*pdata = append(*pdata, sonarConfig{
+		url: string(match[1]),
+		file: checker.File{
+			Path:      pathfn,
+			Type:      checker.FileTypeSource,
+			Offset:    offset,
+			EndOffset: endOffset,
+		},
+	})
+
+	return true, nil
+}
+
+func findLine(content, data []byte) (uint, error) {
```

Comment:
What if there are multiple matching lines?

Reply:
For now it does not matter so I kept it simple. We get the first it's probably enough. I don't know if it's allowed to have multiple ones.